contracts/YakRegistry.sol

```diff
@@ -1,40 +1,79 @@
+    mapping(address => uint256) public strategyIdForStrategyAddress;
+    mapping(address => uint256[]) public strategyIdsForDepositToken;
+    EnumerableSet.AddressSet private strategies;
+
+    struct StrategyInfo {
+        uint256 id;
+        address strategyAddress;
+        bool depositsEnabled;
+        address depositToken;
+        address rewardToken;
+        uint256 minTokensToReinvest;
+        uint256 maxTokensToDepositWithoutReinvest;
+        uint256 adminFeeBips;
+        uint256 devFeeBips;
+        uint256 reinvestRewardBips;
+    }
 
     event AddStrategy(address indexed strategy);
-    event RemoveStrategy(address indexed strategy);
 
     constructor() {}
 
-    function addStrategies(address[] calldata strategies) external onlyOwner {
-        for (uint i = 0; i < strategies.length; i++) {
-            _addStrategy(strategies[i]);
-        }
+    function isActiveStrategy(address _strategy) external view returns (bool) {
+        YakStrategy strategy = YakStrategy(_strategy);
+        return strategies.contains(_strategy) && strategy.DEPOSITS_ENABLED();
+    }
+
+    function strategiesForDepositTokenCount(address _depositToken) external view returns (uint256) {
+        return strategyIdsForDepositToken[_depositToken].length;
+    }
+
+    function strategyInfo(uint256 _sId) external view returns (StrategyInfo memory) {
+        address strategyAddress = strategies.at(_sId);
+        YakStrategy strategy = YakStrategy(strategyAddress);
+        return
+            StrategyInfo({
+                id: _sId,
+                strategyAddress: address(strategy),
+                depositsEnabled: strategy.DEPOSITS_ENABLED(),
+                depositToken: address(strategy.depositToken()),
+                rewardToken: address(strategy.rewardToken()),
+                minTokensToReinvest: strategy.MIN_TOKENS_TO_REINVEST(),
+                maxTokensToDepositWithoutReinvest: strategy.MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST(),
+                adminFeeBips: strategy.ADMIN_FEE_BIPS(),
+                devFeeBips: strategy.DEV_FEE_BIPS(),
+                reinvestRewardBips: strategy.REINVEST_REWARD_BIPS()
+            });
     }
 
-    function removeStrategies(address[] calldata strategies) external onlyOwner {
-        for (uint i = 0; i < strategies.length; i++) {
-            _removeStrategy(strategies[i]);
-        }
+    function strategyId(address _strategy) external view returns (uint256) {
+        return strategyIdForStrategyAddress[_strategy];
     }
 
-    function _addStrategy(address strategy) private {
-        registeredStrategies[strategy] = true;
-        emit AddStrategy(strategy);
+    function strategiesCount() external view returns (uint256) {
+        return strategies.length();
     }
 
-    function _removeStrategy(address strategy) private {
-        registeredStrategies[strategy] = false;
-        emit RemoveStrategy(strategy);
+    function addStrategy(address _strategy) external onlyOwner {
```

Comment:
There is a risk here of adding non-compliant strategy which returns invalid values on views.

Reply:
Added view interface check and returns(StrategyInfo) for owner to double check his newly added entry